docs/en/advanced_tutorials/model_analysis.md

```diff
@@ -0,0 +1,144 @@
+# Model Complexity Analysis
+
+We provide a tool to help the complexity analysis for the network. We borrow the idea from the implementation of [fvcore](https://github.com/facebookresearch/fvcore) to build this tool, and plan to support more custom operators in the future.  Currently, it provides the interfaces to compute "parameter", "activation" and "flops" of the given model, and supports print the related information layer-by-layer in terms of network structure or table. The analysis tool provides both operator-level and module-level flop counts simultaneously. Please refer to [Flop Count](https://github.com/facebookresearch/fvcore/blob/main/docs/flop_count.md) for implementation details of how to accurately measure the flops of one operator if interested.
+
+## What's FLOPs
+
+Flop is not a well-defined metric in complexity analysis, we follow [detectron2](https://detectron2.readthedocs.io/en/latest/modules/fvcore.html#fvcore.nn.FlopCountAnalysis) to use one fused multiple-add as one flop.
+
+## What's Activation
+
+Activation is used to measure the feature quantity produced from one layer.
+
+For example, given the inputs with shape `inputs = torch.randn((1, 3, 10, 10))`, and one linear layer with `conv = nn.Conv2d(in_channels=3, out_channels=10, kernel_size=1)`.
+
+We get the `output` with shape `(1, 10, 10, 10)` after feed the `inputs` into `conv`. The activation quantity of `output` of this `conv` layer is `1000=10*10*10`
+
+## Usage
+
+Let's start with an example as follows:
+
+```python
+import torch
+from torch import nn
+from mmengine.analysis import get_model_complexity_info
+# return a dict of analysis results, including:
+# ['flops', 'flops_str', 'activations', 'activations_str', 'params', 'params_str', 'out_table', 'out_arch']
+
+class InnerNet(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.fc1 = nn.Linear(10,10)
+        self.fc2 = nn.Linear(10,10)
+    def forward(self, x):
+        return self.fc1(self.fc2(x))
+
+
+class TestNet(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.fc1 = nn.Linear(10,10)
+        self.fc2 = nn.Linear(10,10)
+        self.inner = InnerNet()
+    def forward(self, x):
+        return self.fc1(self.fc2(self.inner(x)))
+
+input_shape = (1, 10)
+model = TestNet()
+
+analysis_results = get_model_complexity_info(model, input_shape)
+
+print(analysis_results['out_table'])
+print(analysis_results['out_arch'])
+
+print(analysis_results['flops_str'])
+print(analysis_results['params_str'])
+```
+
```

Comment:
It would be better to use the model defined in https://mmengine.readthedocs.io/en/latest/get_started/15_minutes.html#build-a-model as another example.